ipp-v3-java-devkit/src/main/java/com/intuit/ipp/query/GenerateQuery.java

```diff
@@ -30,87 +30,82 @@
 import com.intuit.ipp.query.expr.NumberPath;
 import com.intuit.ipp.query.expr.StringPath;
 import com.intuit.ipp.util.Logger;
-import net.sf.cglib.proxy.NoOp;
 
 /**
  * Class used to generate the query string
- * 
+ *
  */
 public final class GenerateQuery {
 
 	/**
 	 * logger instance
 	 */
 	private static final org.slf4j.Logger LOG = Logger.getLogger();
-	
+
 	/**
 	 * variable LEN_3
 	 */
 	private static final int LEN_3 = 3;
-	
+
 	/**
 	 * variable path
 	 */
 	public static ThreadLocal<Path<?>> path = new ThreadLocal<Path<?>>();
-	
+
 	/**
 	 * varriable message
 	 */
 	private static QueryMessage message = new QueryMessage();
 
+	/**
+	 * variable CLASSNAME_SPLIT_PATTERN
+	 */
+	private static final String CLASSNAME_SPLIT_PATTERN = "\\$";
+
 	/**
 	 * Constructor to have private modifier as it has only static methods
 	 */
 	private GenerateQuery() {
 	}
 
 	/**
-	 * Method to create the query entity for the given class
-	 * 
+	 *
 	 * @param cl the class
 	 * @return the proxified object
 	 */
 	@SuppressWarnings("unchecked")
-	public static <T> T createQueryEntity(Class<T> cl) {
-		Enhancer enhancer = new Enhancer();
+	public static <T> T createQueryEntity(Class<T> cl) throws InstantiationException, IllegalAccessException {
+		Class<?> proxied = null;
 		if (cl.isInterface()) {
 			LOG.debug("The given class is interface");
-			//enhancer.setInterfaces(new Class[] { cl });
-			//enhancer.setCallback(new MyMethodInterceptor());
 		} else {
-			enhancer.setSuperclass(cl);
+			proxied = new ByteBuddy()
+					.subclass(cl)
+					.method(ElementMatchers.not(ElementMatchers.isClone().or(ElementMatchers.isFinalizer()).or(ElementMatchers.isEquals()).or(ElementMatchers.isHashCode()).or(ElementMatchers.isToString())))
+					.intercept(MethodDelegation.to(new MyMethodInterceptor()))
+					.make()
+					.load(GenerateQuery.class.getClassLoader(), ClassLoadingStrategy.Default.WRAPPER)
+					.getLoaded();
 		}
-		enhancer.setCallbackFilter(CALLBACK_FILTER);
-		enhancer.setCallbacks(new Callback[] {NoOp.INSTANCE, new MyMethodInterceptor()});
-		return (T) enhancer.create();
+		return (T) proxied.newInstance();
```

Comment:
```suggestion
		try {
		    return (T) proxied.newInstance();
		} catch (InstantiationException | IllegalAccessException e) {
		    throw new GenerateQueryException(e);
		}
```
with a new GenerateQueryException type derived from RuntimeException?